### PR TITLE
Add a toggle for network settings in user settings

### PR DESCRIPTION
**Pull request by jaril**
This allows us to toggle the network panel feature from localstorage using the user settings modal.

There's a larger follow up for us to do which would wrap the features in a hook so that when it's toggled, everything updates automatically, i.e. While in the network panel, if you toggle it off, you should be yanked into the console. For now, we've added a `restart required` notice on it to communicate the UI's limitations.

### CODE REVIEW

**Comment by jaril** [comment]
Sorting out a ts error...

File: src/ui/components/SecondaryToolbox/index.tsx [hunk 1]
```diff
@@ -121,6 +138,12 @@ function SecondaryToolbox({
   const showReact = userSettings.showReact;
   const isNode = recordingTarget === "node";
 
+  const allowedPanels = getAllowedPanels(userSettings);
+
+  if (!allowedPanels.includes(selectedPanel)) {
```
Comment (jcmorrow) [inline src/ui/components/SecondaryToolbox/index.tsx:143]: maybe remove the local here, so `getAllowedPanels(userSettings).includes...`

File: src/ui/hooks/settings.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { SettingItemKey } from "ui/components/shared/SettingsModal/types";
 import useAuth0 from "ui/utils/useAuth0";
 import type { UserSettings } from "../types";
 import { ADD_USER_API_KEY, DELETE_USER_API_KEY, GET_USER_SETTINGS } from "ui/graphql/settings";
+import { features } from "ui/utils/prefs";
+import { useEffect, useState } from "react";
```
Comment (jcmorrow) [inline src/ui/hooks/settings.ts:10]: Probably remove these?

**Review by jcmorrow — APPROVED** [review]
Tiny suggestions, looks great! 👍

**Pull request merged**